client/src/app/site/motions/modules/motion-block/components/motion-block-list/motion-block-list.component.ts

```diff
@@ -127,21 +134,29 @@ export class MotionBlockListComponent extends BaseListViewComponent<ViewMotionBl
     }
 
     /**
-     * Helper function reset the form and set the default values
+     * Helper function reset the form and set the default values as well as closing the modal dialog
      */
     public resetForm(): void {
         this.createBlockForm.reset();
         this.createBlockForm.get('agenda_type').setValue(this.defaultVisibility);
+        if (this.dialogRef) {
+            this.dialogRef.close();
+        }
+        this.dialogRef = null;
     }
 
     /**
-     * Click handler for the plus button
+     * Click handler for the plus button.
+     * Opens the dialog for motion block creation.
      */
     public onPlusButton(): void {
-        if (!this.isCreatingNewBlock) {
-            this.resetForm();
-            this.isCreatingNewBlock = true;
-        }
+        this.resetForm();
+        this.dialogRef = this.dialog.open(this.newMotionBlockDialog, {
+            width: '400px',
+            maxWidth: '90vw',
+            maxHeight: '90vh',
+            disableClose: true
+        });
```

Comment:
rather than ussing `onSaveNew` as it is, use ` MatDialogRef.afterClosed`

https://material.angular.io/components/dialog/api